modules/store/src/reducer_creator.ts

```diff
@@ -179,6 +189,56 @@ export function on(
   return { reducer, types };
 }
 
+/**
+ * @description
+ * Creates a reducer function to handle state transitions.
+ *
+ * Reducer creators reduce the explicitness of reducer functions with switch statements.
+ *
+ * @param initialState Provides a state value if the current state is `undefined`, as it is initially.
+ * @param ons Associations between actions and state changes.
+ * @returns A reducer function.
+ *
+ * @usageNotes
+ *
+ * - Must be used with `ActionCreator`'s (returned by `createAction`).  Cannot be used with class-based action creators.
+ * - An action type should only be associated with at most one state change function, similar to switch statements.
+ *   - In the case this is violated, the latest defined associated will be used (the latest `on` function passed).
+ * - The returned `ActionReducer` should additionally be returned from an exported `reducer` function.
```

Comment:
We can also use them with an InjectionToken - see the [example app](https://github.com/ngrx/platform/blob/master/projects/example-app/src/app/reducers/index.ts#L37).
I'm not sure if we should add it here tho, as the notes might get big quickly.

Reply:
Yea, notes could get big.  I tried to just include the essentials.  I can add if you'd like, up to you.

Reply:
For me it's fine, I think most devs will reach for the function instead of creating an injection token.